### PR TITLE
Actually virtualize concurrency page

### PR DESCRIPTION
**Pull request by salazarm**
## Summary & Motivation

The wrapper of the virtualized table container didn't have a max-height so the table was rendering every single key and fetching every single key. To fix this I added a wrapper higher up the tree `<Box flex={{direction: 'column'}} style={{overflow: 'auto', height: '100%'}}>`

I also added search filtering since you can no longer search with the browser's built in search now that the table is virtualized

## How I Tested These Changes
Loaded a page from a customer with ~3,000 concurrency keys and saw the page not crash and burn.

video: https://dagsterlabs.slack.com/archives/C070G3UNNLW/p1715946929035669?thread_ts=1715895539.021559&cid=C070G3UNNLW

### CODE REVIEW

File: js_modules/dagster-ui/packages/ui-core/src/instance/InstanceConcurrency.tsx [hunk 1]
```diff
@@ -387,27 +391,53 @@ export const ConcurrencyLimits = ({
...
+  search,
+}: {
+  onAdd?: () => void;
+} & (
+  | {
+      setSearch: (searchString: string) => void;
+      search: string;
+    }
+  | {setSearch?: never; search?: never}
+)) => {
```
Comment (alangenfeld) [inline js_modules/dagster-ui/packages/ui-core/src/instance/InstanceConcurrency.tsx:410]: nit: probably just something i am not used to but this way of describing types is hard to parse to me

**Review by alangenfeld — APPROVED** [review]

**Review by prha — APPROVED** [review]

File: js_modules/dagster-ui/packages/ui-core/src/instance/InstanceConcurrency.tsx (hunk 1 above)
Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/instance/InstanceConcurrency.tsx:410]: This is to be specific about what are the valid parameter combinations.

Comment (salazarm) [reply js_modules/dagster-ui/packages/ui-core/src/instance/InstanceConcurrency.tsx:410]: (eg. `search` and `setSearch` must both be set, or neither be set, you can't specify just one of them).
